src/bsdfs/dielectric.cpp

```diff
@@ -267,8 +267,8 @@ class SmoothDielectric final : public BSDF<Float, Spectrum> {
         BSDFSample3f bs = dr::zeros<BSDFSample3f>();
         Mask selected_r;
         if (likely(has_reflection && has_transmission)) {
-            selected_r = sample1 <= r_i && active;
-            bs.pdf = dr::select(selected_r, r_i, t_i);
+            selected_r = sample1 <= dr::detach(r_i) && active;
+            bs.pdf = dr::select(selected_r, dr::detach(r_i), dr::detach(t_i));
```

Comment:
You could probably simplify this code as follow:
```suggestion
            bs.pdf = dr::detach(dr::select(selected_r, r_i, t_i));
```